### PR TITLE
Add invocation property to AggregateFunction message for specifying distinct vs all

### PR DESCRIPTION
**Pull request by JamesRTaylor**

### CODE REVIEW

File: proto/substrait/function.proto [hunk 1]
```diff
@@ -78,6 +78,18 @@ message FunctionSignature {
...
+
+    AggregateInvocationType invocation = 16;
+
+    enum AggregateInvocationType {
+      AGGREGATE_INVOCATION_TYPE_UNSPECIFIED = 0;
+
+      // Aggregate function invocation returns all values
+      AGGREGATE_INVOCATION_TYPE_ALL = 1;
+
+      // Aggregate function invocation only returns distinct values
```
Comment (jacques-n) [inline proto/substrait/function.proto:90]: I feel like the "only returns distinct values" here and below are misleading. The function doesn't return distinct values, it returns a calculation that is based on only distinct values.

**Comment by JamesRTaylor** [comment]
I've updated the comment, @jacques-n. I also moved the new invocation property to the right proto message.

**Comment by jacques-n** [comment]
@cpcloud and @jvanstraten , you good with this change? It's not breaking but it is an introduction of a new feature in the spec.

**Review by cpcloud — APPROVED** [review]
LGTM, this will get picked up in the next release on Sunday!

**Comment by jacques-n** [comment]
Thanks @JamesRTaylor !

**Comment by jvanstraten** [comment]
Looks like I'm too late, but looks good to me for what it's worth :)

**Pull request merged**